File: sieve.js
var sieve = function(limit){
	primesBoolList = [];
	primes = [];
	for(var i = 0; i <= limit; i++){
		primesBoolList.push(0);
	}
	len = primesBoolList.length
	for(var i = 2; i < len; i++){
		if(primesBoolList[i] !== 1){
			primes.push(i);
			primesBoolList[i] = 1;
			for(var j = i*i; j < limit + 1; j += i){
				primesBoolList[j] = 1;
			}
		}
	}

	return primes;
};

File: Sieve.java
import java.util.ArrayList;

public class Sieve {
	
	public static void main(String[] args){
		ArrayList<Integer> x = sieve(1000000);
		System.out.println(x.size());
	} 
	
	public static ArrayList<Integer> sieve(int limit){
		ArrayList<Integer> primes = new ArrayList<Integer>();
		
		if (limit < 2){
			return primes;
		}
		
		boolean[] primeBools = new boolean[limit + 1];
		
		for (int i = 2; i < limit+1; i++){
			if (primeBools[i] != true){
				primes.add(i);
				primeBools[i] = true;
				for (int j = i*2; j < limit +1; j += i){
					primeBools[j] = true;
				}
			}
		}
		
		return primes;
		
	}

}
